File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def __init__(self):
        self.ans = []
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root == None:
            return self.ans
        q = deque([root])
        
        while len(q) > 0:
            size = len(q)
            v= []
            for i in range(0,size):
                ele = q.popleft()
                v.append(ele.val)
                if ele.left != None:
                    q.append(ele.left)
                if ele.right != None:
                    q.append(ele.right)
            self.ans.append(v)
        return self.ans
                
        
        
        
        

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
class Solution {
public:
    //val
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == NULL) return {};
        queue<TreeNode*> q;
        q.push(root);
        vector<vector<int>> ans;
        while(!q.empty())
        {
            int sz = q.size();
            vector<int> levelNodes;
            for(int i=0;i<sz;i++)
            {
                auto node = q.front();
                q.pop();
                levelNodes.push_back(node->val);
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);


            }
            ans.push_back(levelNodes);
        }
        return ans;
    }
};